### ISSUE TITLE
out-of-bounds read spotted during compilation

### ISSUE THREAD

**Issue by kevinushey** [fstpackage/fst#122]
Saw this during compilation on the development branch:

```
fstcore_v1/character/character_v1.cpp:512:20: warning: array index 2 is past the end of the array (which contains 2 elements)
      [-Warray-bounds]
    _["meta[2]"] = meta[2]);
                   ^    ~
fstcore_v1/character/character_v1.cpp:282:3: note: array 'meta' declared here
  unsigned int meta[2];
  ^
1 warning generated.
```

See:

https://github.com/fstpackage/fst/blob/1766664afdb30312c6d2ea73be12dfa9eb6dd056/src/fstcore_v1/character/character_v1.cpp#L508-L513

https://github.com/fstpackage/fst/blob/1766664afdb30312c6d2ea73be12dfa9eb6dd056/src/fstcore_v1/character/character_v1.cpp#L282

Removing this line would probably suffice:

https://github.com/fstpackage/fst/blob/1766664afdb30312c6d2ea73be12dfa9eb6dd056/src/fstcore_v1/character/character_v1.cpp#L512

**Comment by MarcusKlik** [comment]
Hi @kevinushey, thanks for reporting that and for your work on RStudio, couldn't live without it!

Indeed, I got an [early warning](#118) from CRAN that the static code analysis of Clang 6.0 turned up some issues. I just fixed them (in 44f5b54) and will submit a patch to CRAN in a few days!

**Comment by MarcusKlik** [comment]
Issue fixed in 44f5b54, will be in next CRAN release
